File: project/jimmer-core/src/test/java/org/babyfish/jimmer/model/Author.java
package org.babyfish.jimmer.model;

import org.babyfish.jimmer.Immutable;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;
import java.util.List;

@Immutable
public interface Author {

    @NotBlank
    @Size(min = 1, max = 50)
    String name();

    List<Book> books();

    @Email
    String email();
}

File: project/jimmer-core-kotlin/src/test/kotlin/org/babyfish/jimmer/kt/model/Author.kt
package org.babyfish.jimmer.kt.model

import org.babyfish.jimmer.Immutable
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Size

@Immutable
interface Author {

    @get:NotBlank
    @get:Size(min = 1, max = 50)
    val firstName: String

    @get:NotBlank
    @get:Size(min = 1, max = 50)
    val lastName: String

    val fullName: String
        get() = "$firstName $lastName"

    val gender: Gender

    @get:Email
    val email: String

    val books: List<Book>
}